extension/background/pageMetadata-contentScript.js

```diff
@@ -41,5 +41,105 @@ this.pageMetadataContentScript = (function() {
     true
   );
 
+  function cleanURL(details) {
+    const globalBlockedParams = [
+      "utm_source",
+      "utm_medium",
+      "utm_campaign",
+      "utm_term",
+      "utm_content",
+    ];
+
+    function getParams(URL) {
+      const splitURL = URL.split("?");
+      if (splitURL.length === 1) {
+        return null;
+      }
+
+      const params = {};
+      const rawParams = URL.split("?")[1].split("&");
+
+      for (let i = 0; i < rawParams.length; i++) {
+        const rawParam = rawParams[i].split("=");
+        params[rawParam[0]] = rawParam[1];
+      }
+
+      return params;
+    }
+
+    function buildURL(baseURL, params) {
+      if (Object.keys(params).length === 0) {
+        return baseURL;
+      }
+
+      let newURL = baseURL + "?";
+
+      for (const key in params) {
+        newURL += key + "=" + params[key] + "&";
+      }
+      newURL = newURL.slice(0, newURL.length - 1);
+
+      return newURL;
+    }
+
+    function getDomain(url) {
+      const arr = url.split("/")[2].split(".");
+
+      if (arr.length > 1) {
+        return arr[arr.length - 2] + "." + arr[arr.length - 1];
+      }
+
+      return null;
+    }
+
+    const baseURL = details.split("?")[0];
+
+    const params = getParams(details);
+    if (params === null) {
+      return;
+    }
+
+    const domain = getDomain(details);
+    if (domain === null) {
+      return;
+    }
+
+    const blockedParams = [];
+    for (const gbp of globalBlockedParams) {
+      if (!gbp.includes("@")) {
```

Comment:
I don't understand what this is doing, none of `globalBlockedParams` seem to have `@` in them?